am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java

```diff
@@ -242,4 +261,82 @@ void whenEmptyListOfResourcesShouldReturnEmptyList() {
+                .data(JsonNodeFactory.instance.objectNode())
+                .build())
+            .access(AccessEnvelope.builder()
+                .permissions(ImmutableMap.of(
+                    JsonPointer.valueOf(PARENT_ATTRIBUTE), ImmutableSet.of(CREATE, READ)))
+                .accessType(AccessType.EXPLICIT)
+                .build())
+            .relationships(ImmutableSet.of(OTHER_ROLE_NAME, ROLE_NAME))
+            .build());
+    }
+
+    @Test
+    void whenExplicitAccessWithSameRelationshipParentChildAttributesWithDiffPermissionsShouldNotMergePermissions() {
+        service.grantExplicitResourceAccess(createGrant(resourceId, ACCESSOR_ID, ROLE_NAME,
+            createPermissions(PARENT_ATTRIBUTE, READ_PERMISSION)));
+        service.grantExplicitResourceAccess(createGrant(resourceId, ACCESSOR_ID, ROLE_NAME,
+            createPermissions(CHILD_ATTRIBUTE, CREATE_PERMISSION)));
+
+        FilteredResourceEnvelope result = service.filterResource(ACCESSOR_ID, ROLE_NAMES, createResource(resourceId));
+
+        assertThat(result).isEqualTo(FilteredResourceEnvelope.builder()
+            .resource(Resource.builder()
+                .id(resourceId)
+                .definition(RESOURCE_DEFINITION)
+                .data(JsonNodeFactory.instance.objectNode())
+                .build())
+            .access(AccessEnvelope.builder()
+                .permissions(ImmutableMap.of(
+                    JsonPointer.valueOf(PARENT_ATTRIBUTE), ImmutableSet.of(READ),
+                    JsonPointer.valueOf(CHILD_ATTRIBUTE), ImmutableSet.of(CREATE)))
+                .accessType(AccessType.EXPLICIT)
+                .build())
+            .relationships(ImmutableSet.of(ROLE_NAME))
+            .build());
+    }
+
+    @Test
+    void whenExplicitAccessWithDifferentRelationshipParentChildAttributeDiffPermissionsShouldMergePermissions() {
+        importerService.addRole(OTHER_ROLE_NAME, RoleType.IDAM, SECURITY_CLASSIFICATION, ACCESS_MANAGEMENT_TYPE);
+
+        service.grantExplicitResourceAccess(createGrant(resourceId, ACCESSOR_ID, ROLE_NAME,
+            createPermissions(PARENT_ATTRIBUTE, READ_PERMISSION)));
+        service.grantExplicitResourceAccess(createGrant(resourceId, ACCESSOR_ID, OTHER_ROLE_NAME,
+            createPermissions(CHILD_ATTRIBUTE, CREATE_PERMISSION)));
+
+        FilteredResourceEnvelope result = service.filterResource(ACCESSOR_ID, ROLE_NAMES, createResource(resourceId));
+
+        assertThat(result).isEqualTo(FilteredResourceEnvelope.builder()
+            .resource(Resource.builder()
+                .id(resourceId)
+                .definition(RESOURCE_DEFINITION)
+                .data(JsonNodeFactory.instance.objectNode())
+                .build())
+            .access(AccessEnvelope.builder()
+                .permissions(ImmutableMap.of(
+                    JsonPointer.valueOf(CHILD_ATTRIBUTE), ImmutableSet.of(CREATE, READ),
+                    JsonPointer.valueOf(PARENT_ATTRIBUTE), ImmutableSet.of(READ)))
```

Comment:
Can we move parent up by one line so we have natural parent->child order.

Reply:
No as this is the way it comes back in the assertion.